src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -3932,20 +3932,24 @@ class StubGenerator: public StubCodeGenerator {
       }
     }
 
-    void gen_loads(Register base) {
-      for (uint i = 0; i < L; i += 1) {
-        __ ld(_regs[i], Address(base, 8 * i));
-      }
+    // generate load for the i'th register
+    void gen_load(uint i, Register base) {
+      assert(i < L, "invalid i: %u", i);
+      __ ld(_regs[i], Address(base, 8 * i));
     }
 
-    // Generate code extracting i-th unsigned word (4 bytes).
-    void get_u32(Register dest, uint i, Register rmask32) {
-      assert(i < 2 * L, "invalid i: %u", i);
+    // add i'th 32-bit integer to dest
+    void addw(const Register dest, uint i, const Register rtmp) {
```

Comment:
Can we give this function another name (Maybe something like `add_u32`) to avoid possible confusion with the `addw` assember function? Or simply inline the code into the caller?